components/core/src/clg/clg.cpp

```diff
@@ -86,13 +86,15 @@ static void print_result_binary (const string& orig_file_path, const Message& co
  * Gets an archive iterator for the given file path or for all files if the file path is empty
  * @param global_metadata_db
  * @param file_path
+ * @param begin_ts
+ * @param end_ts
  * @return An archive iterator
  */
-static GlobalMetadataDB::ArchiveIterator* get_archive_iterator (GlobalMetadataDB& global_metadata_db, const std::string& file_path);
+static GlobalMetadataDB::ArchiveIterator* get_archive_iterator (GlobalMetadataDB& global_metadata_db, const std::string& file_path, epochtime_t begin_ts, epochtime_t end_ts);
 
-static GlobalMetadataDB::ArchiveIterator* get_archive_iterator (GlobalMetadataDB& global_metadata_db, const std::string& file_path) {
+static GlobalMetadataDB::ArchiveIterator* get_archive_iterator (GlobalMetadataDB& global_metadata_db, const std::string& file_path, epochtime_t begin_ts, epochtime_t end_ts) {
```

Comment:
Assuming we create a separate iterator just for filtering by time, then we should add a branch here for handling the case where the user specifies a time range.

Reply:
I've added a branch where I check if the begin_ts and end_ts is supplied by the user.